### PR TITLE
add some func for validation

### PR DESCRIPTION
**Pull request by wolfstudy**
add some func for validation

### CODE REVIEW

File: blockchain/Validation.go [hunk 1]
```diff
@@ -476,12 +480,49 @@ func ReceivedBlockTransactions(pblock *model.Block, state *model.ValidationState
...
+	err := utils.BinarySerializer.PutUint32(fileOut, binary.LittleEndian, uint32(messageStart))
+	if err != nil {
+		panic("the messageStart write failed")
+	}
+	utils.WriteVarInt(fileOut, uint64(size))
+
+	// Write block
+	fileOutPos, err := fileOut.Seek(0, 1)
+	if fileOutPos < 0 || err != nil {
+		panic("WriteBlockToDisk: ftell failed")
```
Comment (ludete) [inline blockchain/Validation.go]: don't use panic

File: blockchain/Validation.go [hunk 2]
```diff
@@ -575,18 +616,57 @@ func FindBlockPos(state *model.ValidationState, pos *DiskBlockPos, nAddSize uint
...
+	file.Seek(int64(offset), 0)
+	for length > 0 {
+		now := 65536
+		if int(length) < now {
+			now = int(length)
+		}
+		// Allowed to fail; this function is advisory anyway.
+		_, err := file.Write(buf[:])
+		if err != nil {
+			panic("the file write failed.")
```
[the remark this answers is not in the document]
Comment (wolfstudy) [reply blockchain/Validation.go]: ok, get it

File: blockchain/Validation.go [hunk 3]
```diff
@@ -575,18 +616,57 @@ func FindBlockPos(state *model.ValidationState, pos *DiskBlockPos, nAddSize uint
...
+		}
+		length -= uint32(now)
+	}
 }
 
 func CheckDiskSpace(nAdditionalBytes uint32) bool {
+	fs := syscall.Statfs_t{}
```
[the remark this answers is not in the document]
Comment (wolfstudy) [reply blockchain/Validation.go]: yes, we should get data path,eg:
```
path := conf.GetDataPath()
fs := syscall.Statfs_t{}
err := syscall.Statfs(path, &fs)
if err != nil {
	return logger.ErrorLog("can not get disk info")
}
```

**Comment by ludete** [comment]
OK

**Pull request merged**